packages/nuxt/src/app/entry.ts

```diff
@@ -30,11 +31,19 @@ if (process.server) {
 
     const nuxt = createNuxtApp({ vueApp, ssrContext })
 
+    async function contextCaller (hooks: HookCallback[], args: any[]) {
+      for (const hook of hooks) {
+        nuxtAppCtx.set(nuxt)
+        await hook(...args)
```

Comment:
I think we might need to use `ctx.call` (otherwise same concurrent ctx sharing problem can happen during promise resolution)

Reply:
okay, using `.call`.